logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -162,33 +163,45 @@ public void open() throws IOException {
         for (int pageNum = headCheckpoint.getFirstUnackedPageNum(); pageNum < headCheckpoint.getPageNum(); pageNum++) {
```

Comment:
hmm, I see. so if the head checkpoint is old and this value is incorrect, it doesn't matter